lib/superform/rails.rb

```diff
@@ -331,10 +335,33 @@ def false_option(&)
         end
 
         protected
+
           def map_options(collection)
             OptionMapper.new(collection)
           end
       end
+
+      class MultipleSelectField < SelectField
+        def template(...)
+          # The HTML specification says when multiple parameter passed to select
+          # and all options got deselected web browsers do not send any value to
+          # server. Unfortunately this introduces a gotcha: if a User model has
+          # many roles and have role_ids accessor, and in the form that edits
+          # roles of the user the user deselects all roles from role_ids
+          # multiple select box, no role_ids parameter is sent.
+          input(type: "hidden", name: dom.name, value: "")
+          super(...)
+        end
+
+        protected
+
+          def field_attributes
+            super.merge(
+              name: "#{dom.name}[]",
```

Comment:
The `field(:foo).collection` method will generate the correct field names for this without having to manually interpolate strings into the field name. Example at https://github.com/rubymonolith/superform/blob/main/README.md?plain=1#L135-L142 and implementation at https://github.com/rubymonolith/superform/blob/main/lib/superform.rb#L224-L259

I haven't thought too much about exactly how this would be integrated, but if you're willing to bounce around a bit with these commits I think we can get there.